src/tg/number_of_principal_components_selection.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+
+from torch.nn.functional import cross_entropy
+from pytorch_lightning.trainer import Trainer
+from pytorch_lightning.callbacks import EarlyStopping, LearningRateMonitor, ModelCheckpoint
+
+from nn.models import MLPClassifier
+from configs.phenotype_config import get_accuracy
+from configs.split_config import FOLDS_NUMBER
+from tg.data_provider import DataProvider
+
+
+LOG_FILE = '/home/genxadmin/accuracy.log'
+
+
+def get_model(num_classes, num_features, trainer=None):
+    parameters = {
+        'nclass': num_classes,
+        'nfeat': num_features,
+        'optim_params': {
+            'name': 'sgd',
+            'lr': 0.1
+        },
+        'scheduler_params': {
+            'rounds': '10000',
+            'epochs_in_round': 1,
+            'name': 'exponential_lr',
+            'gamma': 0.999
+        },
+        'loss': cross_entropy
+    }
+
+    if not trainer:
+        return MLPClassifier(**parameters)
+    else:
+        parameters['checkpoint_path'] = trainer.checkpoint_callback.best_model_path
+        return MLPClassifier.load_from_checkpoint(**parameters)
+
+
+def get_trainer():
+    return Trainer(
+        max_epochs=10000,
+        callbacks=[
+            EarlyStopping(
+                monitor='val_loss',
+                patience=1000,
+                strict=False,
+                verbose=False,
+                mode='min'
+            ),
+            LearningRateMonitor(logging_interval='step'),
+            ModelCheckpoint(monitor='val_loss', mode='min')
+        ],
+        precision=32,
+        weights_summary='full'
+    )
+
+
+def log(num_components, accuracy_train, accuracy_validation, accuracy_test):
+    with open(LOG_FILE, 'a') as log_file:
+        log_file.write(f'{num_components} {accuracy_train} {accuracy_validation} {accuracy_test}\n')
+
+
+def run_experiment(num_components):
```

Comment:
you have this function in several places. if these functions are not very different from each other, maybe you could put similar parts in the same function and import it?